src/serialmonitor/usbDetector.ts

```diff
@@ -14,11 +14,22 @@ import { SerialMonitor } from "./serialMonitor";
 
 export class UsbDetector {
 
+    public static extensionRoot: string;
+
+    public static getInstance(): UsbDetector {
+        if (!UsbDetector._instance) {
+            UsbDetector._instance = new UsbDetector(UsbDetector.extensionRoot);

```

Comment:
Should check the extensionRoot value carefully since there is no any explicit of the sequence of the calling order